src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnectionOptions.cs

```diff
@@ -100,7 +99,7 @@ public CookieContainer Cookies
         /// <summary>
         /// Gets or sets the URL used to send HTTP requests.
         /// </summary>
-        public Uri Url { get; set; }
+        public Uri Url { get; set; } = default!;
```

Comment:
```csharp


var httpConnectionOptions = new HttpConnectionOptions();
var urlString = httpConnectionOptions.Url.ToString();
```

The above throws, right? If so, I think this should be marked nullable. I might feel differently if it was super common for developer to call the HttpConnectionOptions.Url getter and in practice it was always set, but expect most developers only use the setter.

Reply:
The reason I changed it was because of "`!` fatigue", everywhere we access this property internally it will be non-null
https://github.com/dotnet/aspnetcore/blob/e737b6e15b2dea82d91777645a224df498eec6ec/src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnectionFactory.cs#L59-L61

Reply:
I do appreciate trying to avoid "`!` fatigue" but I don't think that's enough of a reason to misrepresent the nullability of a public property. We could add an internal non-nullable property that mirrors Url and throws if it is null as a sanity check to avoid the fatigue.

Reply:
We do this in places where something *could* be null, but for all typical usage it won't be.

e.g. https://github.com/dotnet/aspnetcore/blob/a08aa2dd318956d5b7295561d9423252f6071a52/src/Hosting/Abstractions/src/WebHostBuilderContext.cs

Reply:
Right, that's why I said this initially:

> I might feel differently if it was super common for developer to call the HttpConnectionOptions.Url getter and in practice it was always set, but I expect most developers just call WithUrl() and never touch this property.

Reply:
> I expect most developers just call WithUrl() and never touch this property

Right, that makes it less of an issue if we lie with the API.

Reply:
I would argue that if it's not a huge issue either way, we should default to no lying with the API.

Reply:
I'll try out a private member so we can avoid using `!` everywhere in HttpConnection then.